### ISSUE TITLE
Is it bug in parsing meridiem?

### ISSUE THREAD

**Issue by VyacheslavMik** [andrewmcveigh/cljs-time#112]
Hello. Thanks for your library!

Currently, I can parse such format `hh:mm A` with this value `10:00 T`. I think that you forgot default clause in cond in this function:

```clj
(defn parse-meridiem
  ([]
   (fn [s]
     (let [[[m n] s] (split-at 2 s)
           meridiem (str m n)
           [meridiem s] (cond (#{"am" "pm" "AM" "PM"} meridiem)
                              [meridiem s]
                              (#{\a \p} m)
                              [({\a "am" \p "pm"} m) (cons n s)]
                              (#{\A \P} m)
                              [({\A "am" \P "pm"} m) (cons n s)])]
       [[:meridiem (keyword meridiem)] (string/join s)]))))
```

May be it will be better:

```clj
(defn parse-meridiem
  ([]
   (fn [s]
     (let [[[m n] s] (split-at 2 s)
           meridiem (str m n)
           err #(ex-info
                 (str "Invalid meridiem format: " meridiem) {:type :parse-error})
           [meridiem s] (cond (#{"am" "pm" "AM" "PM"} meridiem)
                              [meridiem s]
                              (#{\a \p} m)
                              [({\a "am" \p "pm"} m) (cons n s)]
                              (#{\A \P} m)
                              [({\A "am" \P "pm"} m) (cons n s)]
                              :default
                              (throw (err)))]
       [[:meridiem (keyword meridiem)] (string/join s)]))))
```

**Comment by VyacheslavMik** [comment]
Also, with this format I can use such values `999T` and it's parsed without errors.

**Comment by andrewmcveigh** [comment]
Thanks for reporting. I think you're analysis is correct, a failure to parse should throw an error.

**Comment by VyacheslavMik** [comment]
Hello, @andrewmcveigh. Thanks for your answer!
Do you fix this or may be create pull request?

**Comment by andrewmcveigh** [comment]
A PR would be awesome 👍

**Comment by VyacheslavMik** [comment]
Hello, @andrewmcveigh. I'm fixed this issues. How quickly you can review pull request?

**Comment by andrewmcveigh** [comment]
Fixed with #113
